File: 2358-maximum-number-of-groups-entering-a-competition/2358-maximum-number-of-groups-entering-a-competition.py
class Solution:
    def maximumGroups(self, grades: List[int]) -> int:
        l,r=0,10**9;
        ans=-1
        while l<=r:
            mid=(l+r)//2
            if mid*(mid+1)/2<=len(grades):
                ans=mid
                l=mid+1
            else:
                r=mid-1
        return ans        
        
        

File: 2358-maximum-number-of-groups-entering-a-competition/2358-maximum-number-of-groups-entering-a-competition.cpp
class Solution {
public:
    int maximumGroups(vector<int>& grades) {
        int len=grades.size();
        int l=0;
        int r=1e9;
        int ans=-1;
        while(l<=r)
        {
            int mid=(l+r)>>1;
            if(1ll*mid*(mid+1)/2<=len)
            {
                ans=mid;
                l=mid+1;
            }
            else
            {
                r=mid-1;
            }
        }
        return ans;
    }
};